src/pages/index.js

```diff
@@ -48,20 +51,50 @@ const cardData = {
 };
 
 function renderCard(cardData) {
-  const card = new Card(cardData, "#card-template", (name, link) => {
-    viewImagePopup.open({ name, link });
+  // const card = new Card(cardData, "#card-template", (name, link) => {
+  //   viewImagePopup.open({ name, link });
+  // });
+  const card = new Card({
+    cardData,
+    cardSelector: "#card-template",
+    handleCardClick,
+    handleDeleteClick,
+    handleLikeClick,
+    handleDeleteClick,
   });
+
   return card.getView();
 }
 
-const section = new Section({
-  items: initialCards,
-  renderer: (cardData) => {
-    const card = renderCard(cardData);
-    section.addItem(card);
-  },
-});
-section.renderItems();
+function handleCardClick(name, link) {
+  viewImagePopup.open({ name, link });
+}
+/* ------------------------ delete card confirmation ------------------------ */
+
+const deleteCardConfirmation = new PopupWithConfirmation("#delete-card-popup");
+
+function handleDeleteClick(card) {
+  deleteCardConfirmation.open();
+  deleteCardConfirmation.setSubmitAction(() => {
+    const id = this.getId();
+    api.deleteCard(id).then(() => {
+      deleteCardConfirmation.close();
+      this.handleDeleteCard();
+    });
+  });
+  deleteCardConfirmation.setEventListeners();
+}
+/* ------------------------------- like button ------------------------------ */
+
+function handleLikeClick(id) {
+  if (this._isLiked) {
+    api.addingLike(id).then(() => {
+      this.handleLikeIcon();
+    });
```

Comment:
Needs correction: don't forget to handle api errors (fix that for every api request)